content/blog/asyncapi_codegen_scst.md

```diff
@@ -0,0 +1,186 @@
+
+If you decide to manually create the document after familiarizing yourself with the specification, don’t worry – you won’t be starting with a blank slate. The AsyncAPI initiative has provided a handy, interactive tool called the [AsyncAPI playground](https://playground.asyncapi.io) to make this easier. On the left side of the playground you can familiarize yourself with the specification and make changes to a real AsyncAPI document, and as you do so the right side of the screen updates to show how the document is parsed into a more human-readable format.
+
+![asyncapi playground for creating microservices using spring cloud stream](/images/posts/asyncapi-codegen_pic-01.webp)
+
+The second way is to use an event portal. Solace PubSub+ Event Portal, for example, allows for architects and developers to collaborate using a GUI to design your event-driven architecture. The team would define the applications that exist in the system, as well as the events that are exchanged and the schemas which define them. Having a catalog of well-organized channels and events for reuse will also save you both time and headaches while collaborating, instead of having to comb through a bunch of files in various locations.
+
+Once the design is in place, PubSub+ Event Portal allows the developer to choose the application they are responsible for developing and download the AsyncAPI document in JSON or YAML.
+
+![pic2](/images/posts/asyncapi-codegen_pic-02.webp)
+
+# **Create Event-Driven Microservices Using Spring Cloud Stream Without Learning Messaging APIs**
+
+Now that we have our AsyncAPI document that describes our application it’s time to develop the application. The AsyncAPI [Code Generator](https://github.com/asyncapi/generator) supports templates to generate code for a variety of different languages and protocols, but for this example we’re going to use the [Spring Cloud Stream template](https://github.com/asyncapi/java-spring-cloud-stream-template). One should note that the template generates a Maven project.
+
+The Spring Cloud Stream framework provides an easy way to get started with event-driven microservices by providing binders that allow the developer to create their microservices without having to learn messaging APIs.
+
+### **Download and Run the AsyncAPI Generator**
+
+The first step is of course to install the AsyncAPI generator itself. If you have NodeJS installed this takes just one easy `npm` command as seen below. You can find the required versions in the [Code Generator](https://github.com/asyncapi/generator) on github.
+
+```bash
+npm install -g @asyncapi/generator
+```
+
+Once you have the generator installed you can run it using the `ag` command. At a minimum you must specify the AsyncAPI document to run it against and the template to use as shown below.
+
+```bash
+ag ~/AsyncApiDocument.yaml @asyncapi/java-spring-cloud-stream-template
```

Comment:
```suggestion
ag https://github.com/asyncapi/asyncapi/blob/2.0.0/examples/2.0.0/streetlights.yml @asyncapi/java-spring-cloud-stream-template
```

better to promote new approach that you can provide remote spec location

Reply:
and maybe you would prefer to make your post bullet-proof and use specific version of the generator and template that you know will still work in few weeks :)